runtime/src/bank.rs

```diff
@@ -3580,13 +3583,11 @@ impl Bank {
             let loaded_transaction = raccs.as_mut().unwrap();
 
             collected_rent += loaded_transaction.rent;
-            self.rewards
-                .write()
-                .unwrap()
-                .append(&mut loaded_transaction.rent_debits.0);
+            rent_debits.push(mem::take(&mut loaded_transaction.rent_debits));
```

Comment:
The story starts here building off what @t-nelson added yesterday: instead of appending transaction-level rent debits to the block rewards, they instead now get returned to the caller and routed though transaction status service